File: app/src/main/java/ru/arhlib/app/news/PostActivity.java
package ru.arhlib.app.news;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.webkit.WebView;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import ru.arhlib.app.MyWebViewClient;
import ru.arhlib.app.R;

public class PostActivity extends AppCompatActivity {

    private String link;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.post_activity);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        Intent intent = getIntent();
        link = intent.getStringExtra("link");
        String title = intent.getStringExtra("title");
        String content = intent.getStringExtra("content");
        String imageUrl = intent.getStringExtra("imageUrl");

        String style = "<style>" + getString(R.string.style_css) + "</style>";
        content = style + "<h1>" + title + "</h1>" + content;

        WebView webView = findViewById(R.id.webview);
        webView.setWebViewClient(new MyWebViewClient());
        webView.loadDataWithBaseURL("https://arhlib.ru/", content, "text/html; charset=UTF-8", "UTF-8", null);

        ImageView imageView = findViewById(R.id.imageView);
        if (imageUrl != null) {
            Picasso.get()
                    .load(imageUrl)
                    .centerCrop()
                    .fit()
                    .placeholder(R.color.colorPlaceholder)
                    .into(imageView);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.post_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            case R.id.share:
                startActivity(new Intent()
                        .setAction(Intent.ACTION_SEND)
                        .putExtra(Intent.EXTRA_TEXT, link)
                        .setType("text/plain"));
                return true;
            case R.id.browser:
                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(link)));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/ru/arhlib/app/news/PostActivity.kt
package ru.arhlib.app.news

import android.content.Intent
import android.graphics.Color
import android.net.Uri
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.isVisible
import com.squareup.picasso.Picasso
import ru.arhlib.app.R
import ru.arhlib.app.browser.CustomTabs
import ru.arhlib.app.browser.WebViewClientWithCustomTabs
import ru.arhlib.app.databinding.PostActivityBinding

class PostActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = PostActivityBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        binding.postTitle.text = getPostTitle()

        binding.webview.webViewClient = WebViewClientWithCustomTabs()
        binding.webview.setBackgroundColor(Color.TRANSPARENT)
        binding.webview.loadDataWithBaseURL("https://arhlib.ru/", getContent(), "text/html; charset=UTF-8", "UTF-8", null)

        if (getImageUrl().startsWith("http")) {
            binding.imageView.isVisible = true

            Picasso.get()
                    .load(getImageUrl())
                    .centerCrop()
                    .fit()
                    .placeholder(R.color.colorPlaceholder)
                    .into(binding.imageView)

            binding.imageView.setOnClickListener {
                CustomTabs.openUrl(this, getSourceImageUrl())
            }
        }
    }

    private fun getPostTitle(): String {
        return intent.getStringExtra("title") ?: ""
    }

    private fun getContent(): String {
        val style = "<style>" + getString(R.string.style_css) + "</style>"
        return style + intent.getStringExtra("content")
    }

    private fun getLink(): String {
        return intent.getStringExtra("link") ?: ""
    }

    private fun getImageUrl(): String {
        return intent.getStringExtra("imageUrl") ?: ""
    }

    private fun getSourceImageUrl(): String {
        return intent.getStringExtra("sourceImageUrl") ?: ""
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.post_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                true
            }
            R.id.share -> {
                startActivity(Intent.createChooser(Intent()
                        .setAction(Intent.ACTION_SEND)
                        .putExtra(Intent.EXTRA_TEXT, getLink())
                        .setType("text/plain"), null))
                true
            }
            R.id.browser -> {
                startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(getLink())))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
